src/abstractdataframe/selection.jl

```diff
@@ -270,20 +272,85 @@ function _fix_existing_columns_for_vector(newdf, df, allow_resizing_newdf, lr, @
     allow_resizing_newdf[] = false
 end
 
-function _add_col_check_copy(df, newdf, col_idx, copycols, @nospecialize(fun), newname, @nospecialize(v))
+function _add_col_check_copy(newdf::DataFrame, df::AbstractDataFrame,
+                             col_idx::Union{Nothing, Int, AbstractVector{Int}, AsTable},
+                             copycols::Bool, @nospecialize(fun),
+                             newname::Symbol, @nospecialize(v))
```

Comment:
`v` supposed to be a vector, right? IIRC `@nospecialize` works with type annotations.

Reply:
you are right - thank you for being so careful, actually we can specialize on `v` I think.